mne/encoding/model.py

```diff
@@ -0,0 +1,154 @@
+
+        Parameters
+        ----------
+        raw : instance of MNE Raw
+            The data on which we want to fit a regression model.
+        events : array, shape (n_events, 3)
+            An MNE events array specifying indices for event onsets
+        event_id : dictionary | None
+            A dictionary of (event_name: event_int) pairs, corresponding to the
+            mapping from event name strings to the 3rd column of events.
+        continuous : array, shape (n_feats, n_times)
+            Continuous input features in the regression.
+        continuous_names : array, shape (n_feats,) | None
+            Names for the input continuous variables.
+        picks : array, shape (n_picks,) | None
+            Indices for channels to use in model fitting. If None, all channels
+            will be fit.
+
+        Attributes
+        ----------
+        coefs_ : array, shape (n_channels, n_features)
+            The coefficients fit on each row of X
+        coef_names : array, shape (n_features,)
+            A list of coefficient names, useful for keeping track of time lags
+        """
+        if y is None:
+            raise ValueError('Must supply an output variable `y`')
+        if X.shape[-1] != y.shape[-1]:
+            raise ValueError('X and y must have last dimension same length.')
+        n_chs, n_times = y.shape
+
+        # Prepare the input feature matrix
+        if self.preproc_x is not None:
+            X = self.preproc_x.fit_transform(X)
+        if self.preproc_y is not None:
+            y = self.preproc_y.fit_transform(y)
+        n_features = X.shape[0]
+
+        # Fit the model and assign coefficients
+        self.coef_ = np.zeros([n_chs, n_features])
+        for ii in range(n_chs):
+            i_y = y[ii][:, np.newaxis]
+            self.est.fit(X.T, i_y)
+            self.coef_[ii, :] = self.est.steps[-1][-1].coef_
+        self.X = X
+        self.y = y
+
+    def predict(self, X, preproc_x=None):
+        """Generate predictions using fit coefficients.
+
+        This uses the `coef_` attribute for predictions.
+        """
+        if preproc_x is not None:
+            X = preproc_x.fit_transform(X)
+        X = self.est._pre_transform(X.T)[0].T
+        preds = np.dot(self.coef_, X)
+        return preds
+
+
+def _check_estimator(est):
+    from sklearn.linear_model import Ridge
+    from sklearn.pipeline import Pipeline
+    if est is None:
+        est = Ridge()
+    elif isinstance(est, string_types):
+        if est not in estimator_dict.keys():
+            raise ValueError("No such solver: {0}".format(est))
+        est = estimator_dict[est]
+
+    # Make sure we have a pipeline
+    if not isinstance(est, Pipeline):
+        est = Pipeline([('est', est)])
+    for imethod in ['fit', 'predict']:
+        if not hasattr(est.steps[-1][-1], imethod):
+            raise ValueError('estimator must have a %s method' % imethod)
```

Comment:
Yeah but then users would have to repeat that trick when making their own estimators right?